rocAL/source/decoders/image/rocjpeg_fused_crop_decoder.cpp

```diff
@@ -0,0 +1,226 @@
+
+    RocJpegChromaSubsampling subsampling;
+    switch (desired_decoded_color_format) {
+        case Decoder::ColorFormat::GRAY:
+            _decode_params_batch[index].output_format = ROCJPEG_OUTPUT_Y;
+            _num_channels = 1;
+            break;
+        case Decoder::ColorFormat::RGB:
+        case Decoder::ColorFormat::BGR:
+            _decode_params_batch[index].output_format = ROCJPEG_OUTPUT_RGB;
+            _num_channels = 3;
+            break;
+    };
+
+    if (rocJpegStreamParse(reinterpret_cast<uint8_t *>(input_buffer), input_size, _rocjpeg_streams[index]) != ROCJPEG_STATUS_SUCCESS) {
+        // std::cerr << "Header decode failed\n";
+        return Status::HEADER_DECODE_FAILED;
+    }
+    if (rocJpegGetImageInfo(_rocjpeg_handle, _rocjpeg_streams[index], &num_components, &subsampling, widths, heights) != ROCJPEG_STATUS_SUCCESS) {
+        // std::cerr << "Header decode failed\n";
+
+        return Status::HEADER_DECODE_FAILED;
+    }
+
+    if (widths[0] < 64 || heights[0] < 64) {
+        // std::cerr << "Header decode failed\n";
+
+        return Status::CONTENT_DECODE_FAILED;
+    }
+
+    std::string chroma_sub_sampling = "";
+    GetChromaSubsamplingStr(subsampling, chroma_sub_sampling);
+    if (subsampling == ROCJPEG_CSS_411 || subsampling == ROCJPEG_CSS_UNKNOWN) {
+        return Status::UNSUPPORTED;
+    }
+
+    *width = widths[0];
+    *height = heights[0];
+    _max_decoded_width = max_decoded_width;
+    _max_decoded_height = max_decoded_height;
+
+    _decode_params = &_decode_params_batch[index];
+
+    if (GetChannelPitchAndSizes(_decode_params_batch[index], subsampling, max_widths, max_heights, channels_size, _output_images[index], channel_sizes)) {
+        return Status::HEADER_DECODE_FAILED;
+        // ERR("Header decode failed\n")
+    }
+    _original_image_width = *actual_width = widths[0];
+    _original_image_height = *actual_height = heights[0];
+
+    return Status::OK;
+}
+
+Decoder::Status FusedCropRocJpegDecoder::decode_info(unsigned char *input_buffer, size_t input_size, int *width, int *height, int *color_comps) {
+    RocJpegChromaSubsampling subsampling;
+    uint8_t num_components;
+    uint32_t widths[4] = {};
+    uint32_t heights[4] = {};
+    if (rocJpegStreamParse(reinterpret_cast<uint8_t *>(input_buffer), input_size, _rocjpeg_streams[0]) != ROCJPEG_STATUS_SUCCESS) {
+        return Status::HEADER_DECODE_FAILED;
+    }
+    if (rocJpegGetImageInfo(_rocjpeg_handle, _rocjpeg_streams[0], &num_components, &subsampling, widths, heights) != ROCJPEG_STATUS_SUCCESS) {
+        return Status::HEADER_DECODE_FAILED;
+    }
+    *width = widths[0];
+    *height = heights[0];
+    // _rocjpeg_image_buff_size += (((widths[0] + 8) &~ 7) * ((heights[0] + 8) &~ 7));
+
+    if (widths[0] < 64 || heights[0] < 64) {
+        return Status::CONTENT_DECODE_FAILED;
+    }
+
+    std::string chroma_sub_sampling = "";
+    GetChromaSubsamplingStr(subsampling, chroma_sub_sampling);
```

Comment:
If you are not using the subsampling string anywhere and u are only using the subsampling var alone, should we call the GetChromaSubsamplingStr func?